src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
@Andrews54757 
> GC works with unrefrenced objects. No one said anything with buffers.
- Everything in Javascript inherit Object.
- The buffers become unreferenced after the buffer has been used for sending.

> I am not a MODERATOR, like you are.
- What's ironic is that he runs an agar-like website. And earning money off of it. Unlike you, sitting there, updating OpenAgar without anybody using it and arguing with people about minor things.

> And I didnt even join Polar's chat for the entire year.
- Pretty sure you mean since 2017.
- The messages you've "intercepted" were public. Anyone can see them if they get a link for polarsbots' guild.

> Also, I am not trying have a competition.
- You're sounding like you made one.

> I am keeping out the noobs.
- Elaborate "noobs". Are they "the guys that place 4000 minions on a 1920x1080 map" or "the guys that report bugs, no matter if they're English or not, and in some way helping out the project"?

> And the code for OpenAgar is right out there for everyone to see isnt it?
- All of them visible? Yes.
- All of them readable? No.
- All of them understandable? No.

> And my method is smart. It solves many problems.
- > Ogar IS NOT an enterprise software to be running on double-grade end hardware.
- Well if you really intend to make it like this then host off a server or two and stress test it to the maximum **with 1000 players**. **I don't care that other users are going to host it, I care about your engagement with other people, which is at a record minimum**.

> How about you provide an alternative solution to run my repo, that is better, and why dont you provide the resources needed to keep it?
- He isn't a "hardcore" programmer like you('re trying to be).

> nothing is out of reach. If something bad can happen, theres a good chance it will (with code).
- 1970's people thought of this and added the Backspace and Delete keys.